src/server/api/routers/github-router.ts

```diff
@@ -0,0 +1,82 @@
+import { z } from "zod";
+import { App } from "octokit";
+import { readFileSync } from "fs";
+
+import {
+    createTRPCRouter,
+    protectedProcedure,
+    publicProcedure,
+} from "~/server/api/trpc";
+
+// GitHub router
+export const gitHubRouter = createTRPCRouter({
+    hello: publicProcedure
+        .input(z.object({ text: z.string() }))
+        .query(({ input }) => {
+            return {
+                greeting: `Hello ${input.text}`,
+            };
+        }),
+
+    // Webhook endpoint
+    makeWebhook: protectedProcedure
+        .input(z.object({
+            repo: z.string()
+        }))
+        .mutation(async ({ ctx, input }) => {
+            const { repo } = input;
+            const owner = ctx.session?.user.name;
+
+            const privatePem = readFileSync("private-key.pem", {
+                encoding: "utf-8",
+            });
+        
+            const app = new App({
+                appId: "892635",
```

Comment:
Could you make this an environment variable

Reply:
on it